CMakeLists.txt

```diff
@@ -220,7 +220,7 @@ execute_process(COMMAND ${CMAKE_CXX_COMPILER} "-Wl,--version" ERROR_QUIET OUTPUT
 # For ld.gold and ld.bfs (the GNU linkers) we optimize hard
 if("${LINKER_VERSION}" MATCHES "GNU gold" OR "${LINKER_VERSION}" MATCHES "GNU ld")
   message(STATUS "Setting linker optimizations")
-  if(NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
+  if(NOT (${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC" OR ${ENABLE_GOLD_LINKER} AND ${BUILD_SHARED_LIBS}))
```

Comment:
@oxidase Mind throwing some `()` around the `AND` statement there?  I always get nervous about order of operations when it's not explicit.